src/main/scala/de/m7w3/signal/controller/UnlockDB.scala

```diff
@@ -26,32 +26,36 @@ object UnlockDB {
       id = "password"
       promptText = "Password"
       prefColumnCount = 10
-      text.onChange{button.disable = text.toString().trim.isEmpty}
+      text.onChange {
+        button.disable = text.toString().trim.isEmpty
+      }
     }
     img.visible = false
     button.disable = true
     button.defaultButtonProperty().bind(password.focusedProperty())
     button.onAction = (a: ActionEvent) => {
       button.disable = true
+      img.visible = false
       Platform.runLater{
         val result: Try[Unit] = for{
           s <- context.tryLoadExistingStore(password.getText(), skipCache = true)
-          data <- Try(s.getRegistrationData)
-        } yield{
+          data <- Try(s.getRegistrationData())
+        } yield {
+            // TODO: don't use Main here, use the passed-in application context
             Main.store = s
             Main.account = AccountHelper(data.userName, data.password)
             val root = ChatsList.load(context)
-          //   Main.stage = new PrimaryStage {
-          //     title = "Welcome"
-          //     scene = new Scene(root)
-          // }
+            button.getScene.setRoot(root)
         }
-        result.failed.foreach(t => {
+        result match {
+          case Success(_) =>
```

Comment:
@mfelsche the success case is useless because the scene will change in that case. That is why before it was `.failed.foreach'